app/scripts/models/auth_brokers/base.js

```diff
@@ -140,18 +146,36 @@ define([
      * Called before confirmation polls to persist any data that is needed
      * for email verification. Useful for storing data that may be needed
      * by the verification tab.
+     *
+     * @param {object} account
+     * @return {promise}
      */
-    persist: function () {
-      return p();
+    persist: function (account) {
```

Comment:
pass in a `verificationInfo`  object to `persist`, it's fine to add `context` to that. Passing in the object will allow each broker to add data that it needs stored,